pkg/api/info.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/els0r/goProbe/pkg/version"
+	"github.com/gin-gonic/gin"
+)
+
+// ServiceInfo summarizes the running service's name, version, and commit. If running in
+// kubernetes, it will also print the name of the pod which returned the API call
+type ServiceInfo struct {
+	Name    string `json:"name"`             // Name: service name. Example: global-query
+	Version string `json:"version"`          // Version: (semantic) version and commit short.  Example: 4.0.0-824f5847
+	Commit  string `json:"commit,omitempty"` // Commit: full git commit SHA. Example: 824f58479a8f326cb350085b3a0e287645e11bc1
+	Pod     string `json:"pod,omitempty"`    // Pod: name of kubernetes pod, if available. Example: global-query-5987cbf795-dvnsl
+}
+
+// ServiceInfoHandler returns a handler that returns the service name, version, and commit
+func ServiceInfoHandler(serviceName string) gin.HandlerFunc {
+	info := &ServiceInfo{
+		Name:    serviceName,
+		Version: version.Short(),
+		Commit:  version.GitSHA,
+	}
+
+	// try to ascertain the running pod's name
+	for _, env := range []string{"POD_NAME", "POD", "PODNAME"} {
```

Comment:
Is that a canonical way / order of obtaining the pod name? What if e.g. both `POD_NAME` and `PODNAME` are set (for whatever reason, just playing devil's advocate here) - is the implemented behavior the intended one?

Reply:
I would argue that the person is then doing something useless in their k8s manifest 😺 . The intention here is that you _could_ provide that information from a pod's metadata when deploying, but by no means have to.